Rolling update (#211)
OpenLiberty/guide-kubernetes-intro: README.adoc

// Scaling a deployment
// ================================================================================================

== Rolling update

> Rolling update refers to continually installing Kubernetes patches without disrupting the availability of the deployed applications. Without continuous updates, a Kubernetes cluster will be susceptible to a denial of a service attack. Update the yaml file as follows to add the `rollingUpdate` configuration. 

Reviewer: ```suggestion
Without continuous updates, a Kubernetes cluster is susceptible to a denial of a service attack. Rolling updates continually install Kubernetes patches without disrupting the availability of the deployed applications. Update the yaml file as follows to add the `rollingUpdate` configuration. 
```

[role="code_command hotspot", subs="quotes"]
----
#Replace the Kubernetes configuration file#
`kubernetes.yaml`
----

kubernetes.yaml
[source, yaml, linenums, role="code_column"]
----
include::finish/kubernetes.yaml[]
----

> The [hotspot=rolling1 hotspot=rolling2]`rollingUpdate` configuration has two attributes `maxUnavailable` and `maxSurge`. The [hotspot=maxUnavailable1 hotspot=maxUnavailable2]`maxUnavailable` attribute refers to the the number of Kubernetes pods that are allowed to be unavailable during the update process. Similarly, the [hotspot=maxSurge1 hotspot=maxSurge2]`maxSurge` attribute refers to the number of additional pods that are allowed to be created during the update process.

Reviewer: ```suggestion
The [hotspot=rolling1 hotspot=rolling2]`rollingUpdate` configuration has two attributes, `maxUnavailable` and `maxSurge`. The [hotspot=maxUnavailable1 hotspot=maxUnavailable2]`maxUnavailable` attribute specifies the the maximum number of Kubernetes pods that can be unavailable during the update process. Similarly, the [hotspot=maxSurge1 hotspot=maxSurge2]`maxSurge` attribute specifies the maximum number of additional pods that can be created during the update process.
```
